python3/httplib2/__init__.py

```diff
@@ -161,7 +161,7 @@ def urlnorm(uri):
 # Cache filename construction (original borrowed from Venus http://intertwingly.net/code/venus/)
 re_url_scheme    = re.compile(br'^\w+://')
 re_url_scheme_s  = re.compile(r'^\w+://')
-re_slash         = re.compile(br'[?/:|]+')
+re_slash         = re.compile(br'\W+')
```

Comment:
This new pattern `\W` will replace few very fine and readable symbols, like `-.` (dash, dot).
Please revise pattern.

In particular, replacing dot with comma (later in safename) is a disaster.

Reply:
Ok, let it be `[^a-zA-Z0-9_-.]`? 

> In particular, replacing dot with comma (later in safename) is a disaster.
Not sure what you mean

Reply:
changing `music.mp3` to `music,mp3` is bad

Reply:
yes, I understand, but I don't see where it happens in the code.

Reply:
I think international characters are fine.

How about `re_unsafe = re.compile(r'[^\w_-.()=!]+')` and later code has to be changed a bit to work with unicode.

Name has to be changed as it's really not about slash anymore.

Reply:
Yes, looks good. 

I saw that originally it was just md5 of the name, why did you decide to switch to `name,md5` format?

Reply:
As commit a46fe4ed3d5340b86299c5ac2ff5e7480d96274d  says 

> Fixed bug# 1597306, cache names are no longer opaque

A readable cache key is definitely better for debugging.

I'm sorry, I take it back on international symbols. Those will surely bring troubles with caches other than filesystem.

`re_unsafe = re.compile(r'[^\w\-_.()=!]+', re.ASCII)`

Reply:
```python
if re_url_scheme_s.match(filename):
    if isinstance(filename,bytes):
```
That's an oxymoron: if `filename` is `bytes`, matching it against `str` pattern will always generate

    TypeError: cannot use a string pattern on a bytes-like object

Reply:
Also, idna encoding doesn't improves readability afaik. There's a `unidecode` lib which is pretty good at converting international symbols into something readable in ascii, but adding an additionaly dependency plus slowing things down calling it - all that down just for the sake of debugging convenience is hardly worth the while. 

I'd suggest skipping that 'idna' step entirely.